backend/src/v2/metadata/client_fake.go

```diff
@@ -93,3 +145,54 @@ func (c *FakeClient) GetOrInsertArtifactType(ctx context.Context, schema string)
 func (c *FakeClient) FindMatchedArtifact(ctx context.Context, artifactToMatch *pb.Artifact, pipelineContextId int64) (matchedArtifact *pb.Artifact, err error) {
 	return nil, nil
 }
+
+// Dummy data will create 2 artifacts that belong to a context with id 1
+// The context has pipeline_root, namespace, and bucket_session_info
+// custom properties.
+func (c *FakeClient) createDummyData() {
+	art1 := &pb.Artifact{
```

Comment:
We are creating this Artifact inside a method called by `clients, manager, _ := initWithExperiment(t)` . The `createDummyData` is down in the stack by level 5.
Later, we are comparing the following code in the same stack level. 
```
			expectedArtifact: &apiv2beta1.Artifact{
				ArtifactId:      "0",
				StorageProvider: "s3",
				StoragePath:     "pipeline/some-pipeline-id/task/key0",
				Uri:             "s3://test-bucket/pipeline/some-pipeline-id/task/key0",
				ShareUrl:        "dummy-signed-url", // defined in object_store_fake
				Namespace:       "test-namespace",
				ArtifactType:    "1",
				ArtifactSize:    123,

				CreatedAt:     timestamppb.New(time.Unix(1, 0)),
				LastUpdatedAt: timestamppb.New(time.Unix(1, 0)),
			},
```
I don't see this as a big problem also I am not suggesting a change in the code but I prefer having what we are creating and comparing in the same stack level. I also know that what was created by `createDummyData` will be transformed in `generateResponseArtifact` to match the code shared above.

Reply:
I agree with you, this would have been my preferred approach. However there is no mlmd client call being exposed by the client interface that would allow us to populate the expected data, which is why it's being done here. In general the lack a proper mlmd mock client has resulted in various portions of the code being untested upstream, and I did not want to add to this tech debt.